File: src/CSharp/89.GrayCode.cs

using System;
using System.Linq;
using System.Text;
using System.Collections.Generic;


namespace PeefyLeetCode.GrayCode
{
public class Solution {
    public IList<int> GrayCode(int n) {
        List<int> ans = new List<int>();
        int count = 1 << n;
        for (int i = 0; i < count; i++){
            ans.Add(i ^ i >> 1);
        }
        return ans;
    }
}
}

File: src/Java/1-100/89.GrayCode.java

import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashMap;
import java.util.Queue;
import java.util.HashSet;
import java.util.LinkedList;


class Solution {
    public List<Integer> grayCode(int n) {
        List<Integer> ans = new ArrayList<>();
        int count = 1 << n;
        for (int i = 0; i < count; i++){
            ans.add(i ^ i >> 1);
        }
        return ans;
    }
}
